code/modules/power/singularity/singularity.dm

```diff
@@ -444,3 +446,25 @@
 	eat()
 	if(prob(1))
 		mezzer()
+
+
+/obj/singularity/HasProximity(atom/movable/AM)
+	if(istype(AM, /obj/item/projectile))
```

Comment:
Exclude `/obj/item/projectile/beam/emitter` from this

Reply:
Note: This is so that you can actually have a functional containment for singulos